tsh status panic

I'm having an issue with _tsh status_ with the latest 14.3.4 version running on WSL2 (Ubuntu 22.04)

```
$ uname -a
Linux PF426J1K 5.15.133.1-microsoft-standard-WSL2 #1 SMP Thu Oct 5 21:02:42 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

$ tsh version
Teleport v14.3.4 git:v14.3.4-0-g16b0227 go1.21.6
Proxy version: 14.3.4
```

```
$ tsh status
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x2aa39ef]

goroutine 1 [running]:
github.com/gravitational/teleport/api/profile.profileFromFile({0xc0017d02a0, 0x1c})
        github.com/gravitational/teleport/api@v0.0.0/profile/profile.go:366 +0x8f
github.com/gravitational/teleport/api/profile.FromDir({0xc0025100a8?, 0x0?}, {0xc002516ec4?, 0xc000b00000?})
        github.com/gravitational/teleport/api@v0.0.0/profile/profile.go:349 +0xca
github.com/gravitational/teleport/lib/client.(*FSProfileStore).GetProfile(0xc002516ec4?, {0xc002516ec4?, 0x1d93601?})
        github.com/gravitational/teleport/lib/client/profile.go:144 +0x2c
github.com/gravitational/teleport/lib/client.(*Store).ReadProfileStatus(0xc000394fc0, {0xc002516ec4?, 0x4eb2980?})
        github.com/gravitational/teleport/lib/client/client_store.go:157 +0xcd
github.com/gravitational/teleport/lib/client.(*Store).FullProfileStatus(0xc000394fc0)
        github.com/gravitational/teleport/lib/client/client_store.go:226 +0x22e
github.com/gravitational/teleport/tool/tsh/common.(*CLIConf).FullProfileStatus(0x0?)
        github.com/gravitational/teleport/tool/tsh/common/tsh.go:3942 +0x2f
github.com/gravitational/teleport/tool/tsh/common.onStatus(0xc00196a000)
        github.com/gravitational/teleport/tool/tsh/common/tsh.go:4295 +0x2a
github.com/gravitational/teleport/tool/tsh/common.Run({0x628dd68, 0xc0017d3640}, {0xc0000740d0?, 0x1, 0x1}, {0x0, 0x0, 0xc00009e6d8?})
        github.com/gravitational/teleport/tool/tsh/common/tsh.go:1354 +0xe6f6
github.com/gravitational/teleport/tool/tsh/common.Main()
        github.com/gravitational/teleport/tool/tsh/common/tsh.go:558 +0x2e5
main.main()
        github.com/gravitational/teleport/tool/tsh/main.go:24 +0xf
```
Other commands work, it's just weird that this, the basic status is not available.

_Originally posted by @joval-sv in https://github.com/gravitational/teleport/discussions/38159_

--- a/api/profile/profile.go
+++ b/api/profile/profile.go
@@ -359,10 +359,15 @@ func profileFromFile(filePath string) (*Profile, error) {
 	if err != nil {
 		return nil, trace.ConvertSystemError(err)
 	}
-	var p *Profile
+	var p Profile
 	if err := yaml.Unmarshal(bytes, &p); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	if p.Name() == "" {
+		return nil, trace.NotFound("invalid or empty profile at %q", filePath)
+	}
+
 	p.Dir = filepath.Dir(filePath)
 
 	// Older versions of tsh did not always store the cluster name in the
@@ -371,7 +376,7 @@ func profileFromFile(filePath string) (*Profile, error) {
 	if p.SiteName == "" {
 		p.SiteName = p.Name()
 	}
-	return p, nil
+	return &p, nil
 }
 
 // SaveToDir saves this profile to the specified directory.

